block/manager.go

```diff
@@ -795,6 +958,19 @@ func (m *Manager) getSignature(header types.Header) (*types.Signature, error) {
 	return &signature, nil
 }
 
+func (m *Manager) getTxsFromBatch() cmtypes.Txs {
+	batch := m.bq.Next()
+	var txs cmtypes.Txs
+	if batch == nil {
+		txs = make(cmtypes.Txs, 0)
+	} else {
+		for _, tx := range batch.Transactions {
+			txs = append(txs, tx)
+		}
+	}
+	return txs
```

Comment:
```suggestion
	batch := m.bq.Next()
	if batch == nil {
		return make(cmtypes.Txs, 0)
	}
	txs := make(cmtypes.Txs, 0, len(batch.Transactions))
	for _, tx := range batch.Transactions {
		txs = append(txs, tx)
	}
	return txs
```
If we allocate the size of the batch then we can avoid reallocating each append.